### PR TITLE
Release prep: Documentation, Changelog, tests, ect.

### PR DESCRIPTION
**Pull request by suckatrash**
Release prep.  Went through steps outlined in [repo_standards](https://github.com/ploperations/repo_standards)

### CODE REVIEW

**Comment by suckatrash** [comment]
@genebean I don't know why these tests are failing, they're pretty standard and work pretty much everywhere else. 

While I was looking into this I realized that the module has a [dependency](https://github.com/ploperations/ploperations-github/blob/master/manifests/listener.pp#L7-L8) on an `apache` class, but not the one form the forge.  Seems we have one in our control repo in the `dist` folder.

I'm sort of questioning whether we should publish this on the forge at this point, thoughts?

**Comment by genebean** [comment]
your spec file has a .pp extension instead of a .rb - that's why its busted

**Comment by genebean** [comment]
we should hold releasing to the forge for now.

File: README.md [hunk 1]
```diff
@@ -1,8 +1,11 @@
 # github
 
-Welcome to your new module. A short overview of the generated parts can be found in the PDK documentation at https://puppet.com/pdk/latest/pdk_generating_modules.html .
+![](https://img.shields.io/puppetforge/pdk-version/ploperations/ploperations-github.svg?style=popout)
+![](https://img.shields.io/puppetforge/v/ploperations/ploperations-github.svg?style=popout)
+![](https://img.shields.io/puppetforge/dt/ploperations/ploperations-github.svg?style=popout)
+[![Build Status](https://travis-ci.org/ploperations/ploperations-ploperations-github.svg?branch=master)](https://travis-ci.org/ploperations/ploperations-ploperations-github)
 
-The README template below provides a starting point with details about what information to include in your README.
+This module mirrors your GitHub repos locally, so that you can rapidly create and destroy repositories before the heat death of the universe!
```
Comment (genebean) [inline README.md:9]: Will you add a block right here until we fix the private dependency:

```markdown

> NOTE: this module currently uses a private fork of the apache module. We are working to migrate it to the normal version of apache and plan to release it on the Forge after that.
```

**Review by genebean — CHANGES_REQUESTED** [review]
When doing docs for puppet strings both a description and a summary are needed... even if they are the same. So... the top of a class file should look kinda like this:

```markdown
# Defined type that creates a github mirror for a given repo
#
# @summary Defined type that creates a github mirror for a given repo
#
# @param ensure
```

**Comment by genebean** [comment]
Have you tested to see if this will work with the public apache module? I saw there is no version in the fixtures file

**Comment by suckatrash** [comment]
Wow.. ok, the tests are working much better now thanks @genebean 
ape

Nope, I sort of assumed there would be problems with the public module but I can try it out.

**Comment by suckatrash** [comment]
Looks like the differences in the `apache` classes have surfaced in these tests.

**Comment by genebean** [comment]
Looks like your fixtures need https://forge.puppet.com/puppetlabs/yumrepo_core as that’s bundled with agents but not the puppet gem

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by genebean** [comment]
Bummer, real failures 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by hsnodgrass** [comment]
@genebean @suckatrash Tests passed. Once this is merged I can add this repo to the control repo's Puppetfile.

**Comment by hsnodgrass** [comment]
@genebean I've update the apache version to use a range.

**Review by genebean — APPROVED** [review]

**Pull request merged**